### PR TITLE
Provide macros for easy logging of string_view/TextView values.

### PR DESCRIPTION
**Pull request by ywkaras**
Using printf-like functions.

### CODE REVIEW

**Comment by ywkaras** [comment]
[approve ci]

File: include/tscore/Diags.h [hunk 1]
```diff
@@ -194,3 +194,17 @@ extern inkcoreapi Diags *diags;
 #define is_diags_on(_t) 0
 
 #endif // TS_USE_DIAGS
+
+#if !defined(TS_SVP)
+
+// Macros for convinient output of string_view (and thus also ts::TextView) instances using printf-like
+// functions.
+//
+#define TS_SVPF "%.*s"
```
Comment (bneradt) [inline include/tscore/Diags.h:203]: This functions kind of like the `PRIuMAX` and related types, right? Maybe call this TS_PRIsv?

**Review by bneradt — CHANGES_REQUESTED** [review]
Thanks Walt. This looks helpful. I'd consider using it.

**Comment by maskit** [comment]
Do you mind if I keep using the standard unsweetened way? I'd rather wait for C++20 and remember std::format format.

**Comment by ywkaras** [comment]
Do we have a target date to switch to C++20?

Using string_view instead of std::string avoids a heap allocation.  The problem is that it's easier to do printf-line logging output with std::string than with string_view.

**Comment by ywkaras** [comment]
Looks like std::format outputs an std::string instance.  What are the advantages over using std::stringstream?

**Comment by maskit** [comment]
I'm not suggesting to use std::format for debug logs. I'm saying I don't want to remember the TS specific format specifiers and the macro just for one less parameter.

**Pull request closed without merging**